[LBSE] Correct visual overflow computation for filters on g

rwlbuis opened the pull request:
#### 44777cb431bb83660de805b629a1a3d589a95b0a
<pre>
[LBSE] Correct visual overflow computation for filters on g
<a href="https://bugs.webkit.org/show_bug.cgi?id=276038">https://bugs.webkit.org/show_bug.cgi?id=276038</a>

Reviewed by Nikolas Zimmermann.

Correct visual overflow computation for filters on g by adding a new
option to SVGBoundingBoxComputation::DecorationOption to calculate the filter
effect for otherwise empty geometries.

* LayoutTests/platform/mac-sonoma-wk2-lbse-text/TestExpectations:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/imported/w3c/web-platform-tests/svg/import/filters-background-01-f-manual-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/imported/w3c/web-platform-tests/svg/import/filters-background-01-f-manual-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/imported/w3c/web-platform-tests/svg/import/filters-tile-01-b-manual-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/imported/w3c/web-platform-tests/svg/import/filters-tile-01-b-manual-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/imported/w3c/web-platform-tests/svg/import/types-dom-01-b-manual-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/imported/w3c/web-platform-tests/svg/import/types-dom-01-b-manual-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/W3C-SVG-1.1/filters-tile-01-b-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/W3C-SVG-1.1/types-basicDOM-01-b-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/W3C-SVG-1.1/types-basicDOM-01-b-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/filters/filter-empty-g-expected.txt:
* Source/WebCore/rendering/svg/SVGBoundingBoxComputation.cpp:
(WebCore::SVGBoundingBoxComputation::adjustBoxForClippingAndEffects const):
(WebCore::SVGBoundingBoxComputation::computeVisualOverflowRect):
* Source/WebCore/rendering/svg/SVGBoundingBoxComputation.h:
(WebCore::SVGBoundingBoxComputation::computeVisualOverflowRect): Deleted.

Canonical link: <a href="https://commits.webkit.org/281016@main">https://commits.webkit.org/281016@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d6b3b07e59cfc2a6331345c51254f58581c63a60

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/58414 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37741 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/10898 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/62040 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/8858 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/60543 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/45377 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/9055 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/47312 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6320 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/60445 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/35352 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50502 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/28156 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/32108 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/7801 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/7862 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/54061 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/8071 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/63743 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/2327 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/8110 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/54632 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/2335 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/50522 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/54698 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12898 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/1971 "Passed tests") | | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/33570 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/34656 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/35740 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/34401 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann commented on Source/WebCore/rendering/svg/SVGBoundingBoxComputation.h:
```diff
@@ -68,7 +69,10 @@ class SVGBoundingBoxComputation {
     {
-        auto repaintBoundingBoxWithoutTransformations = computeDecoratedBoundingBox(renderer, repaintBoundingBoxDecoration | DecorationOption::IncludeOutline | DecorationOption::IgnoreTransformations);
+        DecorationOptions options = repaintBoundingBoxDecoration | DecorationOption::IncludeOutline | DecorationOption::IgnoreTransformations;
+        if (renderer.isRenderSVGContainer())
```
I think we prefer is<RenderSVGContainer>.

rwlbuis replied:
Sure, will fix, thanks for the review!

nikolaszimmermann reviewed (approved):
Looks sane, r=me.

rwlbuis committed:
[LBSE] Correct visual overflow computation for filters on g
https://bugs.webkit.org/show_bug.cgi?id=276038

Reviewed by Nikolas Zimmermann.

Correct visual overflow computation for filters on g by adding a new
option to SVGBoundingBoxComputation::DecorationOption to calculate the filter
effect for otherwise empty geometries.

* LayoutTests/platform/mac-sonoma-wk2-lbse-text/TestExpectations:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/imported/w3c/web-platform-tests/svg/import/filters-background-01-f-manual-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/imported/w3c/web-platform-tests/svg/import/filters-background-01-f-manual-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/imported/w3c/web-platform-tests/svg/import/filters-tile-01-b-manual-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/imported/w3c/web-platform-tests/svg/import/filters-tile-01-b-manual-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/imported/w3c/web-platform-tests/svg/import/types-dom-01-b-manual-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/imported/w3c/web-platform-tests/svg/import/types-dom-01-b-manual-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/W3C-SVG-1.1/filters-tile-01-b-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/W3C-SVG-1.1/types-basicDOM-01-b-expected.png: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/W3C-SVG-1.1/types-basicDOM-01-b-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/filters/filter-empty-g-expected.txt:
* Source/WebCore/rendering/svg/SVGBoundingBoxComputation.cpp:
(WebCore::SVGBoundingBoxComputation::adjustBoxForClippingAndEffects const):
(WebCore::SVGBoundingBoxComputation::computeVisualOverflowRect):
* Source/WebCore/rendering/svg/SVGBoundingBoxComputation.h:
(WebCore::SVGBoundingBoxComputation::computeVisualOverflowRect): Deleted.

Canonical link: https://commits.webkit.org/281016@main
